pkg/compactor/compactor.go

```diff
@@ -198,6 +216,14 @@ func (cfg *Config) Validate() error {
 		return errInvalidShardingStrategy
 	}
 
+	if cfg.ShardingStrategy == util.ShardingStrategyShuffle {
+		if !cfg.ShardingEnabled {
+			return errShardingRequired
+		} else if limits.CompactorTenantShardSize <= 0 {
```

Comment:
Hmmm didn't the document that value of "0" is ok and means disable shuffle sharding (i.e. fallback to default sharding)?

Reply:
Ok, I look at else where and it seems like current code elsewhere do check for `<=0` while having default value `0`.  So most likely this is just following what existing code does. So I think this is a separate problem we can address in separate issue (if it end up being an issue). I will talk to more experienced Cortex maintainer about his.